hw/mcu/microchip/pic32mz/src/pic32mz_periph.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * SPI_0
+ *   SCK1  -> RD1
+ */
+static const struct mips_spi_cfg spi_0_cfg = {
+    .mosi = MYNEWT_VAL(SPI_0_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_0_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTD(1),
+};
+
+/*
+ * SPI_1
+ *   SCK2  -> RG6
+ */
+static const struct mips_spi_cfg spi_1_cfg = {
+    .mosi = MYNEWT_VAL(SPI_1_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_1_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTG(6),
+};
+
+/*
+ * SPI_2
+ *   SCK3  -> B14
+ */
+static const struct mips_spi_cfg spi_2_cfg = {
+    .mosi = MYNEWT_VAL(SPI_2_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_2_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTB(14),
+};
+
+/*
+ * SPI_3
+ *   SCK4  -> RD10
+ */
+static const struct mips_spi_cfg spi_3_cfg = {
+    .mosi = MYNEWT_VAL(SPI_3_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_3_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTD(10),
+};
+
+#ifdef _SPI5_BASE_ADDRESS
+/*
+ * SPI_4
+ *   SCK5  -> RF13
+ */
+static const struct mips_spi_cfg spi_4_cfg = {
+    .mosi = MYNEWT_VAL(SPI_4_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_4_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTF(13),
+};
+#endif
+
+#ifdef _SPI6_BASE_ADDRESS
+/*
+ * SPI_5
+ *   SCK6  -> RD15
+ */
+static const struct mips_spi_cfg spi_5_cfg = {
+    .mosi = MYNEWT_VAL(SPI_5_MASTER_PIN_MOSI),
+    .miso = MYNEWT_VAL(SPI_5_MASTER_PIN_MISO),
+    .sck = MCU_GPIO_PORTD(15),
+};
+#endif
+
+/*
+ * I2C_0 -> I2C1
+ *   SCL1 -> RA14 (D10 for 64 pin packges)
+ *   SDA1 -> RA15 (D9 for 64 pin packges)
+ */
+static const struct mips_i2c_cfg i2c_0_cfg = {
+#if __PIC32_PIN_COUNT > 64
+    .scl = MCU_GPIO_PORTA(14),
+    .sda = MCU_GPIO_PORTA(15),
+#else
+    .scl = MCU_GPIO_PORTD(10),
+    .sda = MCU_GPIO_PORTD(9),
+#endif
+    .frequency = MYNEWT_VAL(I2C_0_FREQ_KHZ) * 1000,
+};
+
+#ifdef _I2C2_BASE_ADDRESS
+/*
+ * I2C_1 -> I2C2
+ *   SCL2 -> RA2
+ *   SDA2 -> RA3
+ */
+static const struct mips_i2c_cfg i2c_1_cfg = {
+    .scl = MCU_GPIO_PORTA(2),
+    .sda = MCU_GPIO_PORTA(3),
+    .frequency = MYNEWT_VAL(I2C_1_FREQ_KHZ) * 1000,
+};
+#endif
+
+/*
+ * I2C_2 -> I2C3
+ *   SCL3 -> RF8 (D3 for 64 pin packges)
+ *   SDA3 -> RF2 (D2 for 64 pin packges)
+ */
+static const struct mips_i2c_cfg i2c_2_cfg = {
+#if __PIC32_PIN_COUNT > 64
+    .scl = MCU_GPIO_PORTF(8),
+    .sda = MCU_GPIO_PORTF(2),
+#else
+    .scl = MCU_GPIO_PORTD(3),
+    .sda = MCU_GPIO_PORTD(2),
+#endif
+    .frequency = MYNEWT_VAL(I2C_2_FREQ_KHZ) * 1000,
+};
+
+/*
+ * I2C_3 -> I2C4
+ *   SCL4 -> RG8
+ *   SDA4 -> RG7
+ */
+static const struct mips_i2c_cfg i2c_3_cfg = {
+    .scl = MCU_GPIO_PORTG(8),
+    .sda = MCU_GPIO_PORTG(7),
+    .frequency = MYNEWT_VAL(I2C_3_FREQ_KHZ) * 1000,
+};
+
+/*
+ * I2C_4 -> I2C5
+ *   SCL5 -> RF5
+ *   SDA5 -> RF4
+ */
+static const struct mips_i2c_cfg i2c_4_cfg = {
+    .scl = MCU_GPIO_PORTF(5),
+    .sda = MCU_GPIO_PORTF(4),
+    .frequency = MYNEWT_VAL(I2C_4_FREQ_KHZ) * 1000,
+};
+
+static void
+pic32mz_periph_create_timer_devs(void)
+{
+    int rc;
+
+    if (MYNEWT_VAL(TIMER_0)) {
```

Comment:
Does this deviate from the usual `#if MYNEWT_VAL(x)` style on purpose?

Reply:
With this style generate code is same but paths that are not included in binary are still check for syntax.
It also eliminates need to `#ifdef` RC or `(void)rc`;
Also static variables would need to be in `#ifdef` block can be left alone and `-gc-section` will take care of them